File: src/main/java/hu/gerviba/webschop/service/CircleMemberService.java
package hu.gerviba.webschop.service;

import javax.transaction.Transactional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import hu.gerviba.webschop.dao.CircleMemberRepository;
import hu.gerviba.webschop.model.CircleMemberEntity;

@Service
@Transactional
public class CircleMemberService {

    @Autowired
    CircleMemberRepository repo;
    
    public CircleMemberEntity getById(Long id) {
        return repo.getOne(id);
    }
    
    public void save(CircleMemberEntity member) {
        repo.save(member);
    }

    public CircleMemberEntity getOne(Long memberId) {
        return repo.getOne(memberId);
    }

    public void delete(CircleMemberEntity cme) {
        repo.delete(cme);
    }
    
}

File: src/main/kotlin/hu/kirdev/schpincer/service/CircleMemberService.kt
package hu.kirdev.schpincer.service

import hu.kirdev.schpincer.dao.CircleMemberRepository
import hu.kirdev.schpincer.model.CircleMemberEntity
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
open class CircleMemberService {

    @Autowired
    private lateinit var repo: CircleMemberRepository

    @Transactional(readOnly = true)
    open fun getById(id: Long): CircleMemberEntity {
        return repo.getOne(id)
    }

    @Transactional(readOnly = false)
    open fun save(member: CircleMemberEntity) {
        repo.save(member)
    }

    @Transactional(readOnly = true)
    open fun getOne(memberId: Long): CircleMemberEntity {
        return repo.getOne(memberId)
    }

    @Transactional(readOnly = false)
    open fun delete(cme: CircleMemberEntity) {
        repo.delete(cme)
    }

}
